src/encord_active/lib/common/project.py

```diff
@@ -0,0 +1,245 @@
+from __future__ import annotations
+
+import itertools
+import json
+import logging
+from functools import partial
+from pathlib import Path
+from typing import Dict, List, Optional, Union
+
+import encord.exceptions
+import yaml
+from encord import Project as EncordProject
+from encord.objects.ontology_structure import OntologyStructure
+from encord.orm.label_row import LabelRow
+from encord.project import LabelRowMetadata
+
+from encord_active.lib.common.utils import (
+    collect_async,
+    download_file,
+    fetch_project_meta,
+    slice_video_into_frames,
+)
+
+logger = logging.getLogger(__name__)
+encord_logger = logging.getLogger("encord")
+encord_logger.setLevel(logging.ERROR)
+
+
+class Project:
+    def __init__(self, project_dir: Path, subset_size: Optional[int] = None):
+        """
+        Fetch project's data from local storage.
+
+        :param project_dir: The root directory of the project.
+        :param subset_size: The number of label rows to fetch. If None then all label rows are fetched.
+        """
+        if not project_dir.exists():
+            raise FileNotFoundError("`project_dir` must point to an existing directory")
+        if not project_dir.is_dir():
+            raise NotADirectoryError("`project_dir` must point to an existing directory")
```

Comment:
I stated here that way to signal the user what is exactly the problem with the provided path: "there is no such directory" or "pointing to wrong place". More or less a shortcut for the user.

Reply:
Yeah, that's what I assumed, but it doesn't make sense to raise a `FileNotFoundError` when we don't find the directory IMO 🤷 

Also, if we still consider them different errors, we should make the message more concrete like "`project_dir` doesn't exist" and "`project_dir` is pointing to a file instead of a directory"